### PR TITLE
add nfs server handling and aborting the task (not whole recipe) when seeing infra failure

### PR DESCRIPTION
**Pull request by ftjahjad**

### CODE REVIEW

File: filesystems/nfs/connectathon/runtest.sh [hunk 1]
```diff
@@ -109,29 +113,55 @@ function checkServers ()
...
+                    echo "================================="
+                    echo "Skipping offline nfs servers:" | tee -a $OUTPUTFILE
+                    for S in $offlineservers; do
+                        echo "   |__"${S} | tee -a $OUTPUTFILE
+                    done
+                    echo ""
+                fi
+            fi
+       else
+          report_result $TEST WARN/ABORTED
```
Comment (emanchado) [inline filesystems/nfs/connectathon/runtest.sh]: We abort if `is_run_bycib` is false?

Comment (veruu) [reply filesystems/nfs/connectathon/runtest.sh]: @emanchado this matches the original implementation for running the test outside of CI

File: filesystems/nfs/connectathon/runtest.sh [hunk 2]
```diff
@@ -93,9 +95,11 @@ function checkServers ()
         if [ $? != 0 ]; then
             echo " ***** Warn: $server not online or unavailable in this domain *****" | tee -a $OUTPUTFILE
             offlineserver=$(expr "$offlineserver" + 1)
+            offlineservers=${offlineservers}"$server "
```
[the remark this answers is not in the document]
Comment (ftjahjad) [reply filesystems/nfs/connectathon/runtest.sh]: On my understanding, this is just a style of bash programming.

File: filesystems/nfs/connectathon/runtest.sh [hunk 3]
```diff
@@ -109,29 +113,55 @@ function checkServers ()
...
+                # Abort the task
+                rstrnt-abort --server $RSTRNT_RECIPE_URL/tasks/$TASKID/status
+                exit 0
+            else
+                # Found online nfs servers
+                echo ""
+                echo "===================================="
+                echo "Testing on online nfs servers:" | tee -a $OUTPUTFILE
+                for S in $servers; do
+                    local server=$(echo $S| cut -f1 -d:)
```
[the remark this answers is not in the document]
Comment (ftjahjad) [reply filesystems/nfs/connectathon/runtest.sh]: should we add  a space since the result is the same?

[the remark this answers is not in the document]
Comment (emanchado) [reply filesystems/nfs/connectathon/runtest.sh]: I think it looks easier to read, but it's really not important. Do whatever you think is best, and feel free to merge. I don't want to delay you for such a small thing!

[the remark this answers is not in the document]
Comment (ftjahjad) [reply filesystems/nfs/connectathon/runtest.sh]: good point, adding an extra space for readable

**Comment by ftjahjad** [comment]
@veruu  Thanks for merging

**Pull request merged**